core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -515,3 +518,98 @@ trait DerivationSchema[R] {
   implicit def gen[T]: Typeclass[T] = macro Magnolia.gen[T]
 
 }
+
+trait TemporalSchema {
+
+  private[schema] abstract class TemporalSchema[T <: Temporal](
+    name: String,
+    description: Option[String]
+  ) extends Schema[Any, T] {
+
+    protected def format(temporal: T): ResponseValue
+
+    override def toType(isInput: Boolean, isSubscription: Boolean): __Type =
+      makeScalar(name, description)
+
+    override def resolve(value: T): Step[Any] =
+      PureStep(format(value))
+  }
+
+  def temporalSchema[A <: Temporal](name: String, description: Option[String] = None)(
+    f: A => ResponseValue
+  ): Schema[Any, A] = new TemporalSchema[A](name, description) {
+    override protected def format(temporal: A): ResponseValue = f(temporal)
+  }
+
+  def temporalSchemaWithFormatter[A <: Temporal](name: String, description: Option[String] = None)(
+    formatter: DateTimeFormatter
+  ): Schema[Any, A] =
+    temporalSchema[A](name, description)(a => StringValue(formatter.format(a)))
+
+  implicit lazy val instantSchema: Schema[Any, Instant] =
+    temporalSchema(
+      "Instant",
+      Some("An instantaneous point on the time-line represented by a standard date time string")
+    )(a => StringValue(a.toString))
+
+  lazy val instantEpochSchema: Schema[Any, Instant] =
+    temporalSchema(
+      "Instant",
+      Some("An instantaneous point on the time-line represented by a numerical millisecond since epoch")
+    )(a => IntValue.LongNumber(a.toEpochMilli))
+
+  implicit lazy val localDateTimeSchema: Schema[Any, LocalDateTime] =
+    localDateTimeSchemaWithFormatter(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+
+  def localDateTimeSchemaWithFormatter(formatter: DateTimeFormatter): Schema[Any, LocalDateTime] =
+    temporalSchemaWithFormatter(
+      "LocalDateTime",
+      Some(s"A date-time without a time-zone in the ISO-8601 calendar system in the format of $formatter")
+    )(formatter)
+
+  implicit lazy val offsetDateTimeSchema: Schema[Any, OffsetDateTime] =
+    offsetDateTimeSchemaWithFormatter(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
+
+  def offsetDateTimeSchemaWithFormatter(formatter: DateTimeFormatter): Schema[Any, OffsetDateTime] =
+    temporalSchemaWithFormatter(
+      "OffsetDateTime",
+      Some(s"A date-time with an offset from UTC/Greenwich in the ISO-8601 calendar system using the format $formatter")
+    )(formatter)
+
+  implicit lazy val zonedDateTimeSchema: Schema[Any, ZonedDateTime] =
+    zonedDateTimeSchemaWithFormatter(DateTimeFormatter.ISO_ZONED_DATE_TIME)
+
+  val localDateTimeEpochSchema: Schema[Any, ZonedDateTime] =
```

Comment:
Is this just a variant that's there if someone doesn't want the default one?

Reply:
Yeah, I'm assuming there are people out there who prefer the byte savings of sending a long as an epoch over the clarity of sending a full string. It is just up to them to make sure they interpret it correctly which is why I made it an opt-in instead of the default.